common/djangoapps/pipeline_mako/templates/mako/inline_js.html

```diff
@@ -1,9 +1,12 @@
+<%page expression_filter="h"/>
+<%! from openedx.core.djangolib.markup import HTML %>
 <script \
-% if async:
+% if context.get('async', default=None):
 async \
 % endif
-if defer:
+% if defer:
 defer \
+% endif
 type="text/javascript" charset="utf-8">
-    ${source | safe}
+    ${HTML(source)}
```

Comment:
Responding to https://github.com/edx/edx-platform/pull/24265#discussion_r443762236, but with the latest changes.
* First, just double checking that `source` wil always actually be trusted JavaScript.
* Second, this is a JavaScript context, but the linter is unable to detect it appropriately because the script tag is broken up with the conditionals.  That means we should choose something from: https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/preventing_xss/preventing_xss.html#mako-invalid-js-filter
* From these options, you'll either want `| n, dump_js_escaped_json}` or `| n, decode.utf8}` from the docs.  The former (e.g. `dump_js_escaped_json`) will do JavaScript escaping. I think you'd want this, unless the source has already been JavaScript escaped.  You just want to ensure that your JavaScript doesn't get double-escaped.
```
>>> from django.utils.html import escapejs
# '&' appropriately single-escaped for JavaScript
>>> escapejs('&')
'\\u0026'
# '&' mistakenly double-escaped
>>> escapejs(escapejs('&'))
'\\u005Cu0026'
```